velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -113,16 +115,36 @@ class AggregateWindowFunction : public exec::WindowFunction {
     auto rawFrameEnds = frameEnds->as<vector_size_t>();
 
     FrameMetadata frameMetadata =
-        fillArgVectors(rawFrameStarts, rawFrameEnds, numRows);
-    if (frameMetadata.fixedFirstRow && frameMetadata.nonDecreasingFrameEnd) {
-      fixedStartAggregation(
+        analyzeFrameValues(rawFrameStarts, rawFrameEnds, numRows);
+
+    if (frameMetadata.incrementalAggregation) {
+      vector_size_t startRow;
+      if (frameMetadata.usePreviousAggregate) {
+        // If incremental aggregation can be resumed from the previous block,
+        // then the argument vectors also can be populated from the previous
+        // frameEnd to the current frameEnd. Only the new values are
+        // required for computing aggregates.
+        startRow = previousFrameMetadata_->lastRow + 1;
+      } else {
+        startRow = frameMetadata.firstRow;
+
+        // This is the start of a new incremental aggregation. So the
+        // aggregate_ function object should be initialized.
+        auto singleGroup = std::vector<vector_size_t>{0};
```

Comment:
nit: this can be static or a member variable to avoid allocating vector repeatedly